File: Ahyeon/Greedy/Container With Most Water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        max_water = 0
        start, last = 0, len(height) - 1
        while start < last:
            if height[start] <= height[last]:
                max_water = max(max_water, height[start]*(last - start))
                start += 1
            else:
                max_water = max(max_water, height[last]*(last - start))
                last -= 1
        return max_water

File: Mingeun/Greedy/ContainerWithMostWater.cpp
class Solution {
public:
    
    int getArea(const vector<int>& height, int start, int end){
        return std::min(height[start], height[end]) * abs(start - end);
    }
    
    int maxArea(vector<int>& height) {
        int maxV = -1, start = 0, end = height.size() - 1;
        int i = start, j = end;
        
        while (i < j){
            if (height[i] > height[j])      j--;
            else if (height[j] >= height[i]) i++;
            
            
            // if (i > height.size() - 1 || i < 0) break;
            
            if (getArea(height, i, j) > getArea(height, start, end)) {
                start = i;
                end = j;
            }
            
            
            
            // cout << "start: " << start << ' ' << "end: " << end << endl;
            // cout << "i: " << i << ' ' << "j: " << j << endl;
        }
        
        return getArea(height, start, end);
    }
};